src/cell.rs

```diff
@@ -0,0 +1,24 @@
+use rand::{thread_rng, Rng};
+
+#[derive(Copy, Clone, PartialEq, Debug)]
+pub struct Pos {
+    pub x: i32,
+    pub y: i32,
+}
+
+#[derive(Clone, PartialEq, Debug)]
+pub struct Cell {
+    pub pos: Pos,
+    pub is_dead: bool,
+}
+
+pub type CellMatrix = Vec<Vec<Cell>>;
+pub type Seeder = Box<dyn Fn(u32, u32) -> bool>;
+
+pub fn apply_cell_rules(neighbors_count: i32, is_dead: bool) -> bool {
+    !matches!((neighbors_count, is_dead), (3, true) | (2 | 3, false))
+}
```

Comment:
It's difficult to understand what this does. Is it returning the new state of a cell according to the neighbors and current state? In that case, does `true` correspond to a "dead", or "alive" state?

Reply:
I had a simple match there, but clippy autocorrect me with that. It really is kinda har to read, I'll change it now